addons/stock/static/src/fields/stock_move_line_x2_many_field.js

```diff
@@ -44,14 +44,14 @@ export class SMLX2ManyField extends X2ManyField {
             ...context,
             single_product: true,
             list_view_ref: "stock.view_stock_quant_tree_simple",
-            search_default_on_hand: this.quantListViewShowOnHandOnly,
-            search_default_in_stock: true,
         };
         const productName = this.props.record.data.product_id.display_name;
         const title = _t("Add line: %s", productName);
         let domain = [
             ["product_id", "=", this.props.record.data.product_id.id],
             ["location_id", "child_of", this.props.context.default_location_id],
+            ["on_hand", "=", this.quantListViewShowOnHandOnly],
+            ["quantity", ">", 0.0],
         ];
```

Comment:
```suggestion
            ["quantity", ">", 0.0],
        ];
        if (this.quantListViewShowOnHandOnly) {
            domain.push(["on_hand", "=", true]);
        }
```
Not putting the search filter isn't the same as forcing the domain to `false` for subcontracting